File: Ch1-Fundamentals/0020-Values-Greater-than-Second-Generalized/main.js
function isNumericArray(arr) {
    for (let i = 0; i < arr.length; i++) {
        if (typeof arr[i] !== 'number') {
            return false;
        }
    }
    return true;
}

function valuesGreaterThanSecond(arr) {
    if (arr.length <= 1) {
        return "\n!!! ERROR: There are one or fewer values in this array.\n";
    }
    
    if (isNumericArray(arr) === true) {
        let secondVal = arr[1];
        let count = 0;
        let newArr = [];
        
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] > secondVal) {
                console.log(`Value at idx ${i} > secondVal: ${arr[i]}`);
                count++;
                newArr.push(arr[i]);
            } else if ( arr[i] === secondVal ) {
                continue;
            }
            // else console.log(` x  Value at idx ${i} < secondVal: ${arr[i]}`);
        }
        
        if (newArr.length > 0) {
            console.log("---------------------------------\n=== New Array:", newArr);
        } else {
            console.log("There are no values greater than the second value...");
        }
        return `---------------------------------\nValues greater than second value: ${count}\n`;
    } else {
        return "\n!!! ERROR: The array must consist of numbers.\n";
    }
}

const arr1 = [1, 3, 5, 7, 9, 13];
console.log(valuesGreaterThanSecond(arr1));

const arr2 = [64, 16, 32, 4, 8, 15];
console.log(valuesGreaterThanSecond(arr2));

const arr3 = [4];
console.log(valuesGreaterThanSecond(arr3));

const arr4 = [2, 6];
console.log(valuesGreaterThanSecond(arr4));

File: Ch1-Fundamentals/0020-Values-Greater-than-Second-Generalized/main.py
def is_numeric_array(arr):
    return all(isinstance(x, (int, float)) for x in arr)

def get_values_greater_than_2nd(arr):
    if len(arr) <= 1:
        return "\n!!! ERROR: There are one or fewer values in this array.\n"
    
    if is_numeric_array(arr):
        second_val = arr[1]
        count = 0
        new_arr = []
        
        for i in range(len(arr)):
            if arr[i] > second_val:
                print(f"Value at idx {i} > second_val: {arr[i]}")
                count += 1
                new_arr.append(arr[i])
            elif arr[i] == second_val:
                continue
            # else:
            #     print(f" x  Value at idx {i} < secondVal: {arr[i]}")
        
        if new_arr:
            print("---------------------------------\n=== New Array:", new_arr)
        else:
            print("There are no values greater than the second value...")
        
        return f"---------------------------------\nValues greater than second value: {count}\n"
        
    else:
        return "\n!!! ERROR: The array must consist of numbers."

arr1 = [1, 3, 5, 7, 9, 13]
print(get_values_greater_than_2nd(arr1))

arr2 = [64, 16, 32, 4, 8, 15]
print(get_values_greater_than_2nd(arr2))

arr3 = [4]
print(get_values_greater_than_2nd(arr3))

arr4 = [2, 6]
print(get_values_greater_than_2nd(arr4))